modules/signals/src/with-linked-state.ts

```diff
@@ -0,0 +1,106 @@
+import { linkedSignal, WritableSignal } from '@angular/core';
+
+import { toDeepSignal } from './deep-signal';
+import {
+  InnerSignalStore,
+  SignalsDictionary,
+  SignalStoreFeature,
+  SignalStoreFeatureResult,
+  StateSignals,
+} from './signal-store-models';
+import { isWritableSignal, STATE_SOURCE } from './state-source';
+import { Prettify } from './ts-helpers';
+
+type LinkedStateResult<
+  LinkedStateInput extends Record<
+    string | symbol,
+    WritableSignal<unknown> | (() => unknown)
+  >
+> = {
+  [K in keyof LinkedStateInput]: LinkedStateInput[K] extends WritableSignal<
+    infer V
+  >
+    ? V
+    : LinkedStateInput[K] extends () => infer V
+    ? V
+    : never;
+};
+
+/**
+ *
+ * @description
+ * Adds linked state slices to a SignalStore.
+ *
+ * @usageNotes
+ * ```typescript
+ * type Option = { id: number; label: string };
+ *
+ * const OptionsStore = signalStore(
+ *   withState({ options: [1, 2, 3] }),
+ *   withLinkedState(({ options }) => ({
+ *     selectedOption: () => options()[0] ?? undefined,
+ *   }))
+ * );
```

Comment:
I think this snippet doesn't need to contain the type Option?
Or, the snippets needs to be updated.

```suggestion
 * const OptionsStore = signalStore(
 *   withState({ options: [1, 2, 3] }),
 *   withLinkedState(({ options }) => ({
 *     selectedOption: () => options()[0],
 *   }))
 * );
```